Sources/System/FileControl.swift

```diff
@@ -0,0 +1,216 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+// Strongly typed, Swifty interfaces to the most common and useful `fcntl`
+// commands.
+
+extension FileDescriptor {
+  /// Get the flags associated with this file descriptor
+  ///
+  /// The corresponding C function is `fcntl` with the `F_GETFD` command.
+  @_alwaysEmitIntoClient
+  public func getFlags() throws -> Flags {
+    try Flags(rawValue: fcntl(.getFlags))
+  }
+
+  /// Set the file descriptor flags.
+  ///
+  /// The corresponding C function is `fcntl` with the `F_SETFD` command.
+  @_alwaysEmitIntoClient
+  public func setFlags(_ value: Flags) throws {
+    _ = try fcntl(.setFlags, value.rawValue)
+  }
+
+  /// Get descriptor status flags.
+  ///
+  /// The corresponding C function is `fcntl` with the `F_GETFL` command.
+  @_alwaysEmitIntoClient
+  public func getStatusFlags() throws -> StatusFlags {
+    try StatusFlags(rawValue: fcntl(.getStatusFlags))
+  }
+
+  /// Set descriptor status flags.
+  ///
+  /// The corresponding C function is `fcntl` with the `F_SETFL` command.
+  @_alwaysEmitIntoClient
+  public func setStatusFlags(_ flags: StatusFlags) throws {
+    _ = try fcntl(.setStatusFlags, flags.rawValue)
+  }
+}
+
+// FIXME: We want a better approach for this one.
+extension FileDescriptor {
+  /// TODO: Flesh out PID work and see if there's a better, common formulation
+  /// of the process-or-group id concept.
+  @frozen
+  public struct PIDOrPGID: RawRepresentable {
```

Comment:
Right, this is a cross-cutting concern between this sketch and how we end up handling processes, hence the TODO. I called this out as an "External", meaning that will have to be figured out (or punted) before this can be merged.

Reply:
I'm going to punt this type and API from the fcntl work, to be resolved when we handle processes, `kill`, etc.